### PR TITLE
Pan and zoom

### PR DESCRIPTION
**Pull request by dqna64**
See https://github.com/devsoc-unsw/structs.sh/pull/654

### CODE REVIEW

**Comment by dqna64** [comment]
Todo:
- [ ] enable pan even outside the visualiser svg
- [ ] enable reset transform (reset pan and zoom)

**Comment by dqna64** [comment]
enabled pan by dragging mouse anywhere within #visualiser-container (workspace)
see image 
Screenshot 2024-06-12 at 04 29 45

File: client/src/components/Visualiser/VisualiserCanvas.tsx [hunk 1]
```diff
@@ -1,116 +1,215 @@
...
+        );
+      }
+    };
+
+    const visualiserWorkspaceEle = document.getElementById(VISUALISER_WORKSPACE_ID);
+    if (visualiserWorkspaceEle) {
+      // https://developer.mozilla.org/en-US/docs/Web/API/Resize_Observer_API
+      new ResizeObserver(handleWorkspaceResize).observe(visualiserWorkspaceEle);
+    }
   }, []);
```
Comment (dqna64) [inline client/src/components/Visualiser/VisualiserCanvas.tsx:198]: Use `ResizeObserver` to listen to workspace resize events and keep `workspaceRect` state up-to-date, which keeps user pan/zoom interacts consistent with workspace size.

**Review by dqna64 — COMMENTED** [review]
Keep pan/zoom consistent on workspace resize

**Comment by dqna64** [comment]
Idea for implementation of "allow reset transform" -- use `IntersectionObserver`

**Review by Dammshine — APPROVED** [review]

**Pull request merged**